pkg/config/config.go

```diff
@@ -73,7 +78,9 @@ func init() {
 
 	// Configuration defaults
 	// Agent
-	BindEnvAndSetDefault("dd_url", "https://app.datadoghq.com")
+	// Don't set a default on 'site' to allow detecting with viper whether it's set in config
+	Datadog.BindEnv("site")
+	Datadog.BindEnv("dd_url")
```

Comment:
IIRC viper leverages the default value to configure the expected type as we set [SetTypeByDefaultValue](https://godoc.org/github.com/spf13/viper#SetTypeByDefaultValue) to true. What could be the side-effects of not setting the type?

Would a default of `""` like some other options work?

Reply:
thanks for mentioning this, wasn't aware of this (and the unit tests aren't using our viper wrapper at the moment). Let me fix the unit tests and give this a try 👍

Reply:
tried this: unfortunately `Datadog.IsSet` will always return `true` as long as a default was set on the config key. So AFAIK, as long as a default value is set on a config option, viper doesn't offer any way to check whether a config option was set by the user, which is a problem in this feature because we want to log a message if both `site` and `dd_url` are set by the user.

Not setting a default does mean that `Datadog.Get("site")` doesn't necessarily return a string though.

Alternative: we continue setting a default of the empty string on `site` and `dd_user`, and don't make a difference anymore between the case where the option isn't set by the user, and the case where the option is set by the user to an empty-string-like value.

Reply:
default type is only used in the generic `Get()` method that we do not use https://github.com/spf13/viper/blob/62edee319679b6ceaec16de03b966102d2dea709/viper.go#L663
I think it should be safe to work without, as long as we explictly call `GetString` as we're used to do